File: backend/parking-domain/src/main/java/com/chatchatabc/parking/domain/model/ParkingRole.java
package com.chatchatabc.parking.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "parking_roles")
@AllArgsConstructor
@NoArgsConstructor
public class ParkingRole {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column
    private String name;

    /**
     * Constructor
     *
     * @param name name of the role
     */
    public ParkingRole(String name) {
        this.name = name;
    }
}

File: backend/src/main/kotlin/com/chatchatabc/parking/domain/model/ParkingRole.kt
package com.chatchatabc.parking.domain.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Data

enum class ParkingRoleNames {
    ROLE_OWNER,
    ROLE_MANAGER,
    ROLE_GUARD
}

@Data
@Entity
@Table(name = "parking_roles")
open class ParkingRole(name: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    open lateinit var id: String

    @Column(unique = true)
    open var name: String = name
}